src/test/scala/uk/gov/nationalarchives/consignmentexport/ValidatorSpec.scala

```diff
@@ -0,0 +1,138 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.time.{LocalDateTime, ZonedDateTime}
+import java.util.UUID
+
+import cats.implicits._
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment.Files.Metadata
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment.{Files, Series, TransferringBody}
+import uk.gov.nationalarchives.consignmentexport.Validator.ValidatedFileMetadata
+
+class ValidatorSpec extends ExportSpec {
+
+  private val completeFileMetadata = Files(
+    UUID.randomUUID(),
+    Metadata(
+      1L.some,
+      LocalDateTime.now().some,
+      "originalPath".some,
+      "foiExemption".some,
+      "heldBy".some,
+      "language".some,
+      "legalStatus".some,
+      "rightsCopyright".some)
+  )
+
+  private def consignment(consignmentId: UUID, metadata: List[Files] = List(completeFileMetadata)): GetConsignment = GetConsignment(
+    consignmentId,
+    ZonedDateTime.now().some,
+    ZonedDateTime.now().some,
+    ZonedDateTime.now().some,
+    Series("series-code".some).some,
+    TransferringBody("tb-code".some).some,
+    metadata
+  )
+
+  "validateConsignmentHasFiles" should "return an error if the consignment has no files" in {
+    val consignmentId = UUID.randomUUID()
+    val validator = Validator(consignmentId)
+    val attempt: Either[Throwable, Files] = validator.validateConsignmentHasFiles(consignment(consignmentId, List()))
+    attempt.left.value.getMessage should equal(s"Consignment API returned no files for consignment $consignmentId")
+  }
+
+  "validateConsignmentHasFiles" should "not return an error if the consignment has files" in {
+    val consignmentId = UUID.randomUUID()
+    val validator = Validator(consignmentId)
+    val attempt: Either[Throwable, Files] = validator.validateConsignmentHasFiles(consignment(consignmentId))
+    attempt.isRight should be(true)
+  }
+
+  "validateConsignmentResult" should "return an error if the consignment data is not defined" in {
+    val consignmentId = UUID.randomUUID()
+    val validator = Validator(consignmentId)
+    val attempt = validator.validateConsignmentResult(none)
+    attempt.left.value.getMessage should equal(s"No consignment metadata found for consignment $consignmentId")
+  }
+
+  "validateConsignmentResult" should "not return an error if the consignment data is defined" in {
+    val consignmentId = UUID.randomUUID()
+    val validator = Validator(consignmentId)
+    val attempt = validator.validateConsignmentResult(consignment(consignmentId).some)
+    attempt.isRight should be(true)
+  }
+
+  "validateFileMetadataNotEmpty" should "not return an error if all of the fields are set" in {
```

Comment:
Test name doesn't match the new method name.

Reply:
Good point, I've updated this and the other ones in that file.